spinalcordtoolbox/scripts/sct_detect_compression.py

```diff
@@ -0,0 +1,309 @@
+            # Note: [slice,] is used to convert int to tuple
+            cr = metrics_agg_merged[slice,]['MEAN(compression_ratio)'] * 100    # to convert to %
+            csa = metrics_agg_merged[slice,]['MEAN(area)']
+            solidity = metrics_agg_merged[slice,]['MEAN(solidity)'] * 100     # to convert to %
+            torsion = metrics_agg_merged[slice,]['Torsion']
+            # Compute compression probability
+            probability = predict_compression_probability(cr, csa, solidity, torsion, disc)
+            compression_category = 'high' if probability > CUT_OFF_HIGH else \
+                'moderate' if probability > CUT_OFF_MODERATE else 'low'
+            compression_df = pd.concat([compression_df,
+                                        pd.DataFrame([{'Disc': disc, 'Axial slice #': slice,
+                                                       'Compression probability': probability,
+                                                       'Compression probability category': compression_category,
+                                                       'Compression ratio (%)': cr, 'CSA (mm2)': csa,
+                                                       'Solidity (%)': solidity, 'Torsion (degrees)': torsion}])],
+                                       ignore_index=True)
+
+    return compression_df
+
+
+def get_disc_slices(fname_disc):
+    """
+    Get axial slice numbers corresponding to the intervertebral discs.
+    :param fname_disc: file with disc labels
+    :return: dictionary with disc labels as keys and corresponding axial slice numbers as values
+    """
+    im_disc = Image(fname_disc).change_orientation('RPI')
+    discs = [int(disc) for disc in np.trim_zeros(np.unique(im_disc.data))]  # get unique disc labels (e.g., 3, 4, 5 etc)
+    # Find axial slices corresponding to the discs
+    # Note: [0] is used because get_slices_from_vertebral_levels returns list
+    disc_slices = {disc: get_slices_from_vertebral_levels(im_disc, disc)[0] for disc in discs}
+
+    return disc_slices
+
+
+def main(argv: Sequence[str]):
+    parser = get_parser()
+    arguments = parser.parse_args(argv)
+    verbose = arguments.v
+    set_loglevel(verbose=verbose, caller_module_name=__name__)
+
+    # Parse arguments
+    fname_seg = get_absolute_path(arguments.s)
+    fname_disc = get_absolute_path(arguments.discfile)
+    angle_correction = bool(arguments.angle_corr)
+
+    # Compute shape metrics per slice
+    metrics_agg_merged = compute_shape_metrics(fname_seg, fname_disc, angle_correction, verbose)
+
+    # Get discs corresponding to the compression
+    disc_slices = get_disc_slices(fname_disc)
+
+    # Process axial slices for each disc to compute compression probability
+    compression_df = process_compression(metrics_agg_merged, disc_slices)
+
+    # Save the results to a CSV file
+    fname_out = splitext(add_suffix(fname_seg, '_compression_results'))[0] + '.csv'
+    compression_df.to_csv(fname_out, index=False)
```

Comment:
After adding cut-offs also for moderate and low probability of compression in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4760/commits/48dbc5fec3b78810343c55d0a2695cd3ef83aab8, the output CSV now looks like:

<img width="700" alt="image" src="https://github.com/user-attachments/assets/f06cddfc-646d-4cfa-a5ab-781ab020a3d9" />